NineAnimator/Models/Anime Source/animeunity.it/AnimeUnity+Anime.swift

```diff
@@ -21,91 +21,85 @@ import Foundation
+                let decoder: JSONDecoder = JSONDecoder.init()
+                let user: SearchResponseAnime = try decoder.decode(SearchResponseAnime.self, from: data_json)
+                let decodedResponse = user
                 let reconstructedAnimeLink = AnimeLink(
-                    title: animeTitle,
+                    title: link.title,
                     link: link.link,
-                    image: animeArtworkUrl,
+                    image: link.image,
                     source: self
                 )
-                
                 // Obtain the list of episodes
-                let episodes = try bowl.select("div.text-center div").reduce(into: [EpisodeLink]()) {
-                    collection, container in
-                    let episodeName = try container.select("a.ep-button").text()
-                    var episodeLink = try container.select("a").attr("href")
-                    episodeLink = episodeLink.replacingOccurrences(of: "'", with: "\'")
-                    if !episodeLink.isEmpty {
-                        let newString = episodeLink.replacingOccurrences(of: "\'", with: "%27")
-                        episodeLink = "https://animeunity.it/" + newString
-                        collection.append(.init(
-                            identifier: episodeLink,
-                            name: episodeName,
-                            server: NASourceAnimeUnity.AnimeUnityStream,
-                            parent: reconstructedAnimeLink
-                            ))
-                    }
+                let episodesList = decodedResponse.to_array.map {
+                    episode -> (EpisodeLink, String)? in
+                    let link_ep = episode.link.replacingOccurrences(of: "\\", with: "").dropLast(4)
+                    return (EpisodeLink(
+                        identifier: String(link_ep),
+                        name: episode.number,
+                        server: NASourceAnimeUnity.AnimeUnityStream,
+                        parent: reconstructedAnimeLink
+                    ), "")
                 }
-                
                 // Information
-                var animeSynopsis = ""
+                let synopsis = try bowl.select("div.description").text()
+                // Attributes
                 var additionalAttributes = [Anime.AttributeKey: Any]()
-                _ = try bowl.select("div.card-body p").compactMap {entry -> String in
-                    let trama = ""
-                    if try entry.text().contains("TRAMA") {
-                        let trama = try entry.text()
-                        animeSynopsis = String(trama.dropFirst(7))
-                    }
-                    if try entry.text().contains("TITOLO") {
-                        let title = try entry.text()
-                        animeTitle = String(title.dropFirst(8))
+                _ = try bowl.select("div.info-item").compactMap {entry -> Void in
```

Comment:
```suggestion
                _ = try bowl.select("div.info-item").compactMap { entry -> Void in
```

Also, consider using `reduce(into:_:)` instead of map, since you're not using the results here.

Reply:
I do not understand how `reduce(into:_:)` works, how I have to change it?

Reply:
Check out this example: https://github.com/SuperMarcus/NineAnimator/blob/1ea990147d/NineAnimator/Models/Anime%20Source/4anime.to/FourAnime%2BFeatured.swift#L36-L51

Reply:
```
var additionalAttributes = try bowl.select("div.info-item").reduce(into: [Anime.AttributeKey:Any]()) { contatiner, entry in
                    if try entry.text().contains("Anno") {
                        let year = try entry.select("small").text()
                        contatiner[.airDate] = year
                    }
                    if try entry.text().contains("Valutazione") {
                        let val = try entry.select("small").text()
                        let rating = (val as NSString).floatValue
                        contatiner[.rating] = rating
                        contatiner[.ratingScale] = Float(10.0)
                    }
                }
```
it works, thanks. I'll push a commit if it is ok